packages/react-native/ReactCommon/react/renderer/components/textinput/platform/ios/react/renderer/components/iostextinput/TextInputEventEmitter.cpp

```diff
@@ -172,6 +172,38 @@ void TextInputEventEmitter::onScroll(const Metrics& textInputMetrics) const {
   });
 }
 
+#if TARGET_OS_OSX // [macOS
+void TextInputEventEmitter::onAutoCorrectChange(
+    const Metrics& textInputMetrics) const {
+//  dispatchTextInputEvent("autoCorrectChange", textInputMetrics);
+  dispatchEvent("spellCheckChange", [textInputMetrics](jsi::Runtime& runtime) {
+    auto payload = jsi::Object(runtime);
+    payload.setProperty(runtime, "enabled", textInputMetrics.autoCorrectEnabled);
+    return payload;
```

Comment:
The reason we can't use the one-line version above is because on the native side, the `SettingsChangedEvent` we expect expects `event.nativeEvent.enabled` to be set (See https://github.com/microsoft/react-native-macos/blob/49ce1d304bbeac3c3844a4ef2087540220e11239/packages/rn-tester/js/examples/TextInput/TextInputExample.ios.js#L338-L345)


We could refactor this to expect `autoCorrectEnabled` and so on, but that would be a breaking change to Paper as well. I'm OK with such a breaking change on our next release as this API has limited use, but I wanted to call that out. I haven't decided if this should be a followup PR or changes in this PR (probably the former).

Reply:
Adding onto this.. I wonder if these props can be upstreamed to iOS. In other words, if iOS"s UITextView has similar grammar and spell check props. I've wanted to make a change to add the prop `writingToolsBehavior` for iOS and macOS for related reasons. 

https://developer.apple.com/documentation/appkit/nstextview/writingtoolsbehavior
https://developer.apple.com/documentation/uikit/uiwritingtoolsbehavior

Reply:
Decided I'll make the breaking change a followup. This will stay as is.

Reply:
Made https://github.com/microsoft/react-native-macos/pull/2299 as the followup breaking change, see the last commit of that PR (the rest are the same as this branch)